pkg/operator/sync_common.go

```diff
@@ -115,12 +116,19 @@ func manageSignerCA(client coreclientv1.SecretsGetter, eventRecorder events.Reco
 			return false, fmt.Errorf("failed to rotate signing CA: %v", err)
 		}
 		if len(rotationMsg) == 0 {
+			metrics.Compute(existingCert.NotAfter)
```

Comment:
I think the only place that expiry should be recorded is at the beginning of this else clause (i.e. after L113). In the event that a new certificate is generated (a rare event), the next reconcile will occur shortly after and guarantee that the expiry is recorded in a timely fashion.

Reply:
As I understand it, at the beginning of else block, it is still not certain if the certificate would be re-generated. That will be certain only after `maybeRotateSigningSecret` is executed. That is one case.

There is another case which is handled in the if block where a certificate is not found in the cluster. A new certificate would be generated in this case too. 

So, I think the best place to compute the metric is after the if/else statement. Sounds good?

Reply:
That could be the only place where expiry time can be computed if there is no need to update expiry time for every reconcile loop, which actually makes sense.